include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -3907,25 +3907,39 @@ DoFCellAccessor<DoFHandlerType, level_dof_access>::get_dof_values(
   ForwardIterator    local_values_end) const
 {
   (void)local_values_end;
-  Assert(this->is_artificial() == false,
-         ExcMessage("Can't ask for DoF indices on artificial cells."));
-  Assert(!this->has_children(), ExcMessage("Cell must be active."));
-  Assert(this->dof_handler != nullptr, typename BaseClass::ExcInvalidObject());
-
-  Assert(static_cast<unsigned int>(local_values_end - local_values_begin) ==
-           this->get_fe().dofs_per_cell,
-         typename DoFCellAccessor::ExcVectorDoesNotMatch());
-  Assert(values.size() == this->get_dof_handler().n_dofs(),
-         typename DoFCellAccessor::ExcVectorDoesNotMatch());
-
-  const types::global_dof_index *cache =
+  // Assert(this->is_artificial() == false,
+  //       ExcMessage("Can't ask for DoF indices on artificial cells."));
+  // Assert(!this->has_children(), ExcMessage("Cell must be active."));
+  // Assert(this->dof_handler != nullptr, typename
+  // BaseClass::ExcInvalidObject());
+
+  // Assert(static_cast<unsigned int>(local_values_end - local_values_begin) ==
+  //         this->get_fe().dofs_per_cell,
+  //       typename DoFCellAccessor::ExcVectorDoesNotMatch());
+  // Assert(values.size() == this->get_dof_handler().n_dofs(),
+  //       typename DoFCellAccessor::ExcVectorDoesNotMatch());
```

Comment:
This is of course something we need to work on to improve - assuming that we still need to go over the `DoFCellAccessor` class in the end. One thing I'm thinking of is to enable the assertions in case `level_dof_access` is `false`, and in case it is true we could check
```c++
  Assert(this->level_subdomain_id() != numbers::artificial_subdomain_id,
         ExcMessage("Can't ask for DoF indices on artificial cells."));
  Assert(this->dof_handler != nullptr, typename BaseClass::ExcInvalidObject());
  Assert(static_cast<unsigned int>(local_values_end - local_values_begin) ==
           this->get_fe().dofs_per_cell,
         typename DoFCellAccessor::ExcVectorDoesNotMatch());
  Assert(values.size() == this->get_dof_handler().n_dofs(this->level()),
         typename DoFCellAccessor::ExcVectorDoesNotMatch());
```
(As you can see, I do think that we should be able to keep the third and fourth assertion.) Can you try if something along those lines works? For that to work, we would of course need to have level iterators for the level output, and active iterators for the standard output.